### PR TITLE
ALC-17 Add Transaction batcher and endpoint

### PR DESCRIPTION
**Pull request by iceman2224**
This PR adds the transactions endpoint and a few configurable settings to control how transactions are batched:

transactionBatchMaximum - Maximum batch size to trigger a send

transactionMaxWaitSeconds - Maximum amount of time to wait if a maximum batch size has not been reached

The transaction should contain all information needed to order and batch to miners.  Schema defined in ITransaction.ts

When a transaction has successfully been dispatched to miners, socket.io will alert the clients with a message and a TransactionId

### CODE REVIEW

**Review by diredeaths — APPROVED** [review]
Looks great. We need to find a better place to define the event handlers but its fine for testing purposes.

File: src/services/transactions.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+const MAX_BATCH_SIZE = config.transactionBatchMaximum
+const FORCE_BATCH_MILLIS = config.transactionMaxWaitSeconds * 1000
+@Service()
+export default class TransactionsService {
+  private queue: any
+  private awaitingJobs: any[]
+  private processorLocked: boolean
+  private lastProcessedBatch: number
+
```
Comment (diredeaths) [inline src/services/transactions.ts:23]: The private definitions here can be moved to the constructor
```
constructor(
    @Inject('eventEmitter') private events: EventEmitter,
    @Inject('logger') private logger: Logger,
    private ethers: EthersService,
    private queue: any,
    private awaitingJobs: any[],
    private processorLocked: boolean,
    private lastProcessedBatch: number,
  )
```

**Review by diredeaths — APPROVED** [review]

File: src/services/transactions.ts (hunk 1 above)
Comment (iceman2224) [reply src/services/transactions.ts:23]: Hmm, yeah when I tried that I got an error with typedi
`Service with "MaybeConstructable<Array>" identifier was not found in the container`
So I moved the non-injected dependencies out of the constructor.  Maybe it has to do with the 'any[]' type

Comment (iceman2224) [reply src/services/transactions.ts:19]: changing it to `any` type fixed the issue

**Pull request merged**